### PR TITLE
Priorityq

### PR DESCRIPTION
**Pull request by jaredscarr**

### CODE REVIEW

**Comment by joelstanner** [comment]
Try to avoid committing code with a message of "?". Your coworkers will thank you.

**Comment by joelstanner** [comment]
You have no setup.py package file in this repository. Tox won't work without it.

File: priority_queue.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            most_huge = right
+        if most_huge != self.container[index][0]:
+            value = self.container[index - 1]
+            self.container[index - 1] = self.container[most_huge - 1]
+            self.container[most_huge - 1] = value
+            self.sort_to_leaf(most_huge - 1)
+
+    def peek(self):
+        """Returns the highest priority item in the queue."""
+        return self.container[0]
```
Comment (joelstanner) [inline priority_queue.py:47]: I'm trying to insert tuples into this priority queue. That seems to work, but pop() isn't working. It occasionally throws IndexError. Something is broken here.

File: test_priority_queue.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+# .insert(item): inserts an item into the queue.
+def test_insert():
+    """Test if value is inserted into the queue."""
+    from priority_queue import PriorityQueue
+    pq = PriorityQueue()
+    item = (1, 'value')
+    pq.insert(item)
+    assert pq.container[0] == (1, 'value')
```
Comment (joelstanner) [inline test_priority_queue.py:12]: There isn't much being tested here. Try and thing of more ways to test that insert is working correctly, and make sure to cover edge cases.
